api/envoy/api/v2/route/route.proto

```diff
@@ -243,13 +244,22 @@ message Route {
 // multiple upstream clusters along with weights that indicate the percentage of
 // traffic to be forwarded to each cluster. The router selects an upstream cluster based on the
 // weights.
-// [#comment:next free field: 11]
+// [#comment:next free field: 4]
 message WeightedCluster {
+  // [#comment:next free field: 12]
   message ClusterWeight {
     // Name of the upstream cluster. The cluster must exist in the
     // :ref:`cluster manager configuration <config_cluster_manager>`.
     string name = 1 [(validate.rules).string.min_bytes = 1];
 
+    // [#not-implemented-hide:]
+    // If specified, indicates a server to contact to get the cluster data via CDS.
+    // By default, Envoy will know where to get CDS data from via its bootstrap file, but
+    // other clients (e.g., gRPC) may not have a default.
+    // [#next-major-version: In the v3 API, we should replace the name and config_source
+    // fields with a single ClusterSource field.]
+    envoy.api.v2.core.ConfigSource config_source = 11;
```

Comment:
See my reply to Harvey below.  I understand that Envoy would not currently use this, but gRPC would, and I think that there may be federation cases where this will be useful for Envoy in the future.